aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,162 @@
+```
+
+Note the version, `2.0.0`, reflects the version of the shared runtime that the target app uses. The shared runtime is shown in the *\*.runtimeconfig.json* file. In the sample app, the shared runtime is specified in the *HostingStartupSample.runtimeconfig.json* file.
+
+**Set environment variables**
+
+Set the following environment variables in the context of the app that uses the feature.
+
+ASPNETCORE\_HOSTINGSTARTUPASSEMBLIES
+
+Only hosting startup assemblies are scanned for the `HostingStartupAttribute`. The assembly name of the implementation is provided in this environment variable. The sample app sets this value to `StartupDiagnostics`.
+
+The value can also be set using the [Hosting Startup Assemblies](xref:fundamentals/hosting#hosting-startup-assemblies) host configuration setting.
+
+DOTNET\_ADDITIONAL\_DEPS
+
+The location of the implementation's *\*.deps.json* file.
+
+If the file is placed in the user profile's *.dotnet* folder for per-user use:
+
+```
+<DRIVE>\Users\<USER>\.dotnet\x64\additionalDeps\
+```
+
+If the file is placed in the .NET Core installation for global use, provide the full path to the file:
+
+```
+<DRIVE>\Program Files\dotnet\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\<FEATURE_ASSEMBLY_NAME>.deps.json
+```
+
+The sample app sets this value to:
+
+```
+%UserProfile%\.dotnet\x64\additionalDeps\StartupDiagnostics\
+```
+
+For examples of how to set environment variables for various operating systems, see [Working with multiple environments](xref:fundamentals/environments).
+
+## Discover loaded hosting startup assemblies
+
+To discover hosting startup assemblies loaded by the app or libraries added to the app, read the value of the [HostingStartupAssembliesKey](/dotnet/api/microsoft.aspnetcore.hosting.webhostdefaults.hostingstartupassemblieskey). The sample app reads the key into a `string` array and displays the result in the app's Index page:
+
+[!code-csharp[Main](ihostingstartup/sample/HostingStartupSample/Pages/Index.cshtml.cs?name=snippet1&highlight=14-16)]
+
+## Disable automatic loading of hosting startup assemblies
+
+There are two ways to disable the automatic loading of hosting startup assemblies:
+
+* Set the [Prevent Hosting Startup](xref:fundamentals/hosting#prevent-hosting-startup) host configuration setting.
+* Set the `ASPNETCORE_preventHostingStartup` environment variable.
+
+When either the host setting or the environment variable is set to `true` or `1`, hosting startup assemblies aren't automatically loaded. If both are set, the host setting controls the behavior.
+
+It isn't currently possible to selectively disable a hosting startup assembly added by a library unless the library offers its own configuration option. Disabling hosting startup assemblies using the host setting or environment variable disables them globally and may disable several features of an app.
```

Comment:
want to link currently to the 2.1 bug that added this feature?
https://github.com/aspnet/Hosting/pull/1243

Reply:
We don't generally discuss 2.1 features in the 2.0-era docs. @Rick-Anderson Do you want to make an exception here? I can say something to the effect that 'a future release will offer this feature' and then link that issue. I believe we have done that a few times in the past.

Reply:
@guardrex  yes